include/gui/application.hpp

```diff
@@ -180,23 +179,38 @@ namespace Toolbox {
 
     class FileDialog {
     public:
-        static FileDialog* Instance() {
+        FileDialog() = default;
+        ~FileDialog() {
+            if (m_thread_initialized) {
+                m_thread.join();
+            }
+        }
+
+        static FileDialog *Instance() {
             static FileDialog _instance;
             return &_instance;
         }
-        void OpenDialog(std::filesystem::path starting_path,
-                        GLFWwindow* parent_window,
+        void OpenDialog(std::filesystem::path starting_path, GLFWwindow *parent_window,
                         bool is_directory = false,
-                        std::optional<std::vector<std::pair<
-                        const char*, const char*>>> maybe_filters = std::nullopt);
+                        std::optional<std::vector<std::pair<std::string, std::string>>>
+                            maybe_filters = std::nullopt);
```

Comment:
I was using const char* to try to convey that these are read-only strings that can be statically allocated string literals, and won't get copied all over the place. In Rust this is the kind of thing you do to convey ownership information and resource usage, and technically it's more efficient, but maybe that's too low-level of information for this API.

Reply:
Well the problem I see with const char * usage is that most often things get passed as an std::string in modern C++ since it's an frequent use for anything string related. It is true that there is a bit of overhead, but ultimately it is negligible for something like this which realistically will only get processed N times per dialog initialization. If this were in an active processing loop I'd definitely share the concern though.

When it comes to performance, C++20 (?) introduced std::string_view, which acts as a minimal view of a const char * in its purest essence, and it practically free to construct and pass as copy arguments. It's intended for use with std::string where it makes sense.